### PR TITLE
adding regression test for modules

### LINKED ISSUE #1445: Add code coverage for injecting modules on command line
We are not testing `buildtest build --modules` and `buildtest build --unload-modules` in the test coverage. This can be found in https://app.codecov.io/gh/buildtesters/buildtest/blob/devel/buildtest/builders/base.py under following section.

Screen Shot 2023-04-18 at 3 58 42 PM

In addition we might also want to test for module purge ``buildtest build --module-purge``

### PR DESCRIPTION
**Pull request by kingxleyy**

### CODE REVIEW

File: tests/cli/test_build.py [hunk 1]
```diff
@@ -196,6 +196,41 @@ def test_build_buildspecs(self):
...
+                    numnodes=None,
+                    modules=None,
+                    modulepurge=None,
+                    unload_modules=None,
+                    rerun=None,
+                    executor_type=None,
+                    timeout=None,
+                    limit=None,
+                    save_profile=None,
+                    profile=None,
```
Comment (shahzebsiddiqui) [inline tests/cli/test_build.py]: why are you defining a `BuildTest` class this doesn't make sense. Just call the class and create an object

```python
cmd = Buildtest(...)
cmd.build()
```

File: tests/cli/test_build.py [hunk 2]
```diff
@@ -196,6 +196,7 @@ def test_build_buildspecs(self):
             exclude_buildspecs=[excluded_buildspecs[0]],
             buildtest_system=self.system,
         )
+        cmd = Buildtest(...)
```
Comment (shahzebsiddiqui) [inline tests/cli/test_build.py]: i dont mean you literally copy the suggestion. The `...` was meant for you to fill the appropriate arguments to the class so you can invoke the BuildTest class. Please see examples from above. Once you are ready for review please ping me. In the meantime, i will mark this as a `Draft`

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]

**Comment by Mendi03** [comment]
The regression test is not passing because there is no value for `buildspec_paths` for the code you added.

File: tests/cli/test_build.py [hunk 3]
```diff
@@ -209,6 +209,14 @@ def test_build_buildspecs(self):
             )
             cmd.build()
 
+    def test_module(self):
+        cmd = BuildTest(
+                configuration=configuration,
+                buildspecs=[buildspec_paths],
```
Comment (Mendi03) [inline tests/cli/test_build.py]: The regression test is failing because the variable `buildspec_paths` is empty.

**Review by Mendi03 — CHANGES_REQUESTED** [review]
Check my comment on why the regression is failing. Furthermore, you still need to invoke the class **_with_** a value for `module` and `unload_modules`.

File: tests/cli/test_build.py
Comment (Mendi03) [inline tests/cli/test_build.py:1]: see my previous comment

**Review by shahzebsiddiqui — APPROVED** [review]

**Pull request merged**